packages/content/src/enrichments.ts

```diff
@@ -0,0 +1,443 @@
+import {
+  getItemData,
+  getItemGroups,
+  getUser,
+  IGetUserOptions
+} from "@esri/arcgis-rest-portal";
+import {
+  IHubContent,
+  IHubRequestOptions,
+  IEnrichmentErrorInfo,
+  cloneObject,
+  getProp,
+  getItemHomeUrl,
+  getItemApiUrl,
+  getItemDataUrl,
+  getItemThumbnailUrl,
+  includes
+} from "@esri/hub-common";
+import { getContentMetadata } from "./metadata";
+
+interface IMetadataPaths {
+  updateFrequency: string;
+  metadataUpdateFrequency: string;
+  metadataUpdatedDate: string;
+  reviseDate: string;
+  pubDate: string;
+  createDate: string;
+}
+
+/**
+ * The possible values for updateFrequency
+ *
+ * @enum {string}
+ */
+export enum UpdateFrequency {
+  Continual = "continual",
+  Daily = "daily",
+  Weekly = "weekly",
+  Fortnightly = "fortnightly",
+  Monthly = "monthly",
+  Quarterly = "quarterly",
+  Biannually = "biannually",
+  Annually = "annually",
+  AsNeeded = "as-needed",
+  Irregular = "irregular",
+  NotPlanned = "not-planned",
+  Unknown = "unknown",
+  Semimonthly = "semimonthly"
+}
+
+enum DatePrecision {
+  Year = "year",
+  Month = "month",
+  Day = "day",
+  Time = "time"
+}
+
+function getMetadataPath(identifier: keyof IMetadataPaths) {
+  // NOTE: i have verified that this will work regardless of the "Metadata Style" set on the org
+  const metadataPaths: IMetadataPaths = {
+    updateFrequency:
+      "metadata.metadata.dataIdInfo.resMaint.maintFreq.MaintFreqCd.@_value",
+    reviseDate: "metadata.metadata.dataIdInfo.idCitation.date.reviseDate",
+    pubDate: "metadata.metadata.dataIdInfo.idCitation.date.pubDate",
+    createDate: "metadata.metadata.dataIdInfo.idCitation.date.createDate",
+    metadataUpdateFrequency:
+      "metadata.metadata.mdMaint.maintFreq.MaintFreqCd.@_value",
+    metadataUpdatedDate: "metadata.metadata.mdDateSt"
+  };
+  return metadataPaths[identifier];
+}
+
+function getValueFromMetadata(
+  content: IHubContent,
+  identifier: keyof IMetadataPaths
+) {
+  const path = getMetadataPath(identifier);
+  return path && getProp(content, path);
+}
+/**
+ * Parses an ISO8601 date string into a date and a precision.
+ * This is because a) if somone entered 2018, we want to respect that and not treat it as the same as 2018-01-01
+ * and b) you cannot naively call new Date with an ISO 8601 string that does not include time information
+ * For example, when I, here in mountain time, do new Date('2018').getFullYear() I get "2017".
+ * This is because when you do not provide time or timezone info, UTC is assumed, so new Date('2018') is 2018-01-01T00:00:00 in UTC
+ * which is actually 7 hours earlier here in mountain time.
+ *
+ * @param {string} isoString
+ * @return { date: Date, precision: DatePrecision }
+ */
+export function parseISODateString(isoString: string) {
+  isoString = `${isoString}`;
```

Comment:
For some reason I _thought_ our lint rules complained about argument reassignment.This is probably fine, but might be better practice to not mutate/reassign args.

Reply:
Yeah, seems to me our rules got a little looser a while back, I assume when we added prettier